src/child/actions/window.js

```diff
@@ -3,18 +3,19 @@ import configService from '../../shared/ConfigService';
 import createActionCreator from '../utils/createActionCreator';
 import currentWindowService from '../services/currentWindowService';
 
-export const minimise = createActionCreator(() => ({
+export const minimize = createActionCreator(() => ({
     type: ACTION_TYPES.MINIMIZE
 }));
 
-export const compact = createActionCreator(() => ({
+export const compact = createActionCreator((isMaximised) => ({
     type: ACTION_TYPES.TOGGLE_COMPACT,
-    state: true
+    isCompact: true,
+    previousMaximisedState: isMaximised
```

Comment:
Still need to change to American spelling of maximised/maximized. Hopefully a quick find and replace will sort it :smile: